src/main/kotlin/no/nav/syfo/sykmeldingstatus/kafka/SykmeldingStatusKafkaMessageMapper.kt

```diff
@@ -10,9 +11,104 @@ import no.nav.syfo.model.sykmeldingstatus.ShortNameDTO
+import no.nav.syfo.sykmeldingstatus.api.v1.SykmeldingStatusEventDTO
+import no.nav.syfo.sykmeldingstatus.api.v2.SykmeldingUserEvent
+import no.nav.syfo.sykmeldingstatus.toStatusEvent
+import java.time.OffsetDateTime
+
+fun SykmeldingUserEvent.tilSykmeldingStatusKafkaEventDTO(timestamp: OffsetDateTime, sykmeldingId: String, arbeidsgiver: Arbeidsgiverinfo?): SykmeldingStatusKafkaEventDTO {
+    return SykmeldingStatusKafkaEventDTO(
+        sykmeldingId, timestamp, this.toStatusEvent().tilStatusEventDTO(),
+        arbeidsgiver?.let {
+            ArbeidsgiverStatusDTO(
+                orgnummer = it.orgnummer,
+                juridiskOrgnummer = it.juridiskOrgnummer,
+                orgNavn = it.navn,
+            )
+        },
+        toSporsmalSvarListe(arbeidsgiver),
+    )
+}
+
+fun SykmeldingUserEvent.toSporsmalSvarListe(arbeidsgiver: Arbeidsgiverinfo? = null): List<SporsmalOgSvarDTO> {
+    return listOfNotNull(
+        arbeidssituasjonSporsmalBuilder(),
+        fravarSporsmalBuilder(),
+        periodeSporsmalBuilder(),
+        nyNarmesteLederSporsmalBuilder(arbeidsgiver),
+        forsikringSporsmalBuilder(),
+    )
+}
+
+private fun SykmeldingUserEvent.arbeidssituasjonSporsmalBuilder(): SporsmalOgSvarDTO {
+    return SporsmalOgSvarDTO(
+        tekst = arbeidssituasjon.sporsmaltekst,
+        shortName = ShortNameDTO.ARBEIDSSITUASJON,
+        svartype = SvartypeDTO.ARBEIDSSITUASJON,
+        svar = arbeidssituasjon.svar.name,
+    )
+}
+
+private fun SykmeldingUserEvent.fravarSporsmalBuilder(): SporsmalOgSvarDTO? {
+    if (harBruktEgenmelding != null) {
+        return SporsmalOgSvarDTO(
+            tekst = harBruktEgenmelding.sporsmaltekst,
+            shortName = ShortNameDTO.FRAVAER,
+            svartype = SvartypeDTO.JA_NEI,
+            svar = harBruktEgenmelding.svar.name,
+        )
+    }
+    return null
+}
+
+private fun SykmeldingUserEvent.periodeSporsmalBuilder(): SporsmalOgSvarDTO? {
+    if (egenmeldingsperioder != null) {
+        return SporsmalOgSvarDTO(
+            tekst = egenmeldingsperioder.sporsmaltekst,
+            shortName = ShortNameDTO.PERIODE,
+            svartype = SvartypeDTO.PERIODER,
+            svar = objectMapper.writeValueAsString(egenmeldingsperioder.svar),
+        )
+    }
+    return null
+}
+
+private fun SykmeldingUserEvent.nyNarmesteLederSporsmalBuilder(arbeidsgiver: Arbeidsgiverinfo?): SporsmalOgSvarDTO? {
+    if (arbeidsgiver?.aktivtArbeidsforhold == false) {
+        return SporsmalOgSvarDTO(
+            tekst = "Skal finne ny nærmeste leder",
+            shortName = ShortNameDTO.NY_NARMESTE_LEDER,
+            svartype = SvartypeDTO.JA_NEI,
+            svar = "NEI",
+        )
+    }
+
+    if (nyNarmesteLeder != null) {
+        return SporsmalOgSvarDTO(
+            tekst = nyNarmesteLeder.sporsmaltekst,
+            shortName = ShortNameDTO.NY_NARMESTE_LEDER,
+            svartype = SvartypeDTO.JA_NEI,
+            svar = nyNarmesteLeder.svar.name,
```

Comment:
Kan denne være nei? Det føles tryggere hvis vi bare setter den til JA :)

Reply:
Kan man ikke svare Nei i frontend? isåfall trenger man kanskje ikke spørmålet?

Reply:
den kan vel være nei hvis bruker har sagt nei?

Reply:
Hvis arbeidsforholdet er aktivt så kan denne være både JA eller NEI